src/main/java/gyro/google/storage/BucketResource.java

```diff
@@ -0,0 +1,468 @@
+            bucket.setCors(getCors().stream().map(BucketCors::toBucketCors).collect(Collectors.toList()));
+            bucket.setBilling(getBilling() == null ? null : getBilling().toBucketBilling());
+            bucket.setEncryption(getEncryption() == null ? null : getEncryption().toBucketEncryption());
+            bucket.setIamConfiguration(getIamConfiguration() == null ? null : getIamConfiguration().toBucketIamConfiguration());
+            bucket.setLifecycle(getLifecycle() == null ? null : getLifecycle().toLifecycle());
+            bucket.setLogging(getLogging() == null ? null : getLogging().toBucketLogging());
+            bucket.setRetentionPolicy(getRetentionPolicy() == null ? null : getRetentionPolicy().toBucketRententionPolicy());
+            bucket.setStorageClass(getStorageClass());
+            bucket.setVersioning(getVersioning() == null ? null : getVersioning().toBucketVersioning());
+            bucket.setWebsite(getWebsite() == null ? null : getWebsite().toBucketWebsite());
+
+            // Lock retention policy. This can not be undone and ALL assets must reach policy time to delete bucket.
+            if (changedFieldNames.contains("retention-policy")
+                    && Boolean.TRUE.equals(bucket.getRetentionPolicy().getIsLocked())) {
+                storage.buckets().lockRetentionPolicy(getName(), bucket.getMetageneration()).execute();
+            }
+
+            storage.buckets().update(getName(), bucket).execute();
+        } catch (GoogleJsonResponseException e) {
+            throw new GyroException(e.getDetails().getMessage());
+        }
+    }
+
+    @Override
+    public void delete(GyroUI ui, State state) throws Exception {
+        try {
+            Storage storage = creatClient(Storage.class);
+            storage.buckets().delete(getName()).execute();
+        } catch (IOException e) {
+            throw new GyroException(String.format("Unable to delete Bucket: %s, Google error: %s", getName(), e.getMessage()));
+        }
+    }
+
+    @Override
+    public List<ValidationError> validate() {
+        List<ValidationError> errors = new ArrayList<>();
+
+        if (getName() != null && NAME_PATTERN.matcher(getName()).find()) {
+            errors.add(new ValidationError(this, "name", "Invalid name format."));
+        }
+
+        if (getLabels() != null) {
+            for (Map.Entry<String, String> entry : getLabels().entrySet()) {
+                String key = entry.getKey();
+                String value = entry.getValue();
+                Matcher keyMatcher = LABEL_PATTERN.matcher(key);
+                Matcher valueMatcher = LABEL_PATTERN.matcher(value);
+                if ((key.length() > 63) || (value.length() > 63) || keyMatcher.find() || valueMatcher.find()) {
+                    errors.add(new ValidationError(
+                            this,
+                            "labels",
+                            String.format("Invalid key/value => '%s:%s'", key, value)));
```

Comment:
Same as above.

Reply:
Done. I've added the format rules to the error message.